fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/context/BatchContextAdapter.java

```diff
@@ -0,0 +1,189 @@
+
+import com.ibm.fhir.operation.bulkdata.config.ConfigurationFactory;
+import com.ibm.fhir.operation.bulkdata.model.type.BulkDataContext;
+import com.ibm.fhir.operation.bulkdata.model.type.OperationFields;
+
+/**
+ * Adapts the JobContext or the StepContext
+ *
+ * @implNote There is annoying error in the batch with JWT's cdi
+ * This avoids the error, and consistently treats the job parameters. Previously
+ * we had multiple patterns of reading parameters.
+ */
+public class BatchContextAdapter {
+    private Properties props = null;
+
+    public BatchContextAdapter(StepContext stepCtx) {
+        this.props = stepCtx.getProperties();
+    }
+
+    public BatchContextAdapter(JobContext jobCtx) {
+        this.props = jobCtx.getProperties();
+    }
+
+    public BatchContextAdapter(Properties props) {
+        this.props = props;
+    }
+
+    public BulkDataContext getStepContextForPatientExportPartitionMapper() {
+        BulkDataContext ctx = new BulkDataContext();
+        context(ctx);
+        source(ctx);
+        ctx.setFhirResourceType(props.getProperty(OperationFields.FHIR_RESOURCE_TYPE));
+        return ctx;
+    }
+
+
+    public BulkDataContext getStepContextForSystemChunkReader() {
+        BulkDataContext ctx = new BulkDataContext();
+        context(ctx);
+        source(ctx);
+        audit(ctx);
+        search(ctx);
+        return ctx;
+    }
+
+    public BulkDataContext getStepContextForPatientChunkReader() {
+        BulkDataContext ctx = new BulkDataContext();
+        context(ctx);
+        source(ctx);
+        audit(ctx);
+        search(ctx);
+        ctx.setPartitionResourceType(props.getProperty(OperationFields.PARTITION_RESOURCETYPE));
+        format(ctx);
+        return ctx;
+    }
+
+    public BulkDataContext getStepContextForGroupChunkReader() {
+        BulkDataContext ctx = new BulkDataContext();
+        context(ctx);
+        source(ctx);
+        audit(ctx);
+        search(ctx);
+        ctx.setPartitionResourceType(props.getProperty(OperationFields.PARTITION_RESOURCETYPE));
+        ctx.setGroupId(props.getProperty(OperationFields.FHIR_SEARCH_PATIENT_GROUP_ID));
+        format(ctx);
+        return ctx;
+    }
+
+    public BulkDataContext getJobContextForExportJobListener() {
+        BulkDataContext ctx = new BulkDataContext();
+        context(ctx);
+        source(ctx);
+        ctx.setDataSourcesInfo(props.getProperty(OperationFields.FHIR_DATA_SOURCES_INFO));
+        return ctx;
+    }
+
+    public BulkDataContext getStepContextForSystemChunkWriter() {
+        BulkDataContext ctx = new BulkDataContext();
+        context(ctx);
+        source(ctx);
+        audit(ctx);
+        systemWrite(ctx);
+        format(ctx);
+        return ctx;
+    }
+
+    public BulkDataContext getStepContextForFastResourceWriter() {
+        BulkDataContext ctx = new BulkDataContext();
+        context(ctx);
+        source(ctx);
+        audit(ctx);
+        search(ctx);
```

Comment:
I don't know if we need to set the search-related parameters or not (it doesn't support _typefilter), but we *definitely* need to set the systemWrite ones so that we get our cosPathPrefix.
```suggestion
        systemWrite(ctx);
        search(ctx);
```

Reply:
I'm also confused about why cosPathPrefix is associated with system export...shouldn't we be setting that for *all* exports?  Did you do something different to set it in the Group and Patient exports?